pkg/morph/deploy/notary.go

```diff
@@ -869,3 +878,353 @@ func makeUnsignedDesignateCommitteeNotaryTx(roleContract *rolemgmt.Contract, com
 
 	return tx, nil
 }
+
+// newCommitteeNotaryActor returns notary.Actor builds and sends Notary service
+// requests witnessed by the specified committee members to the provided
+// Blockchain. Given local account pays for transactions.
+func newCommitteeNotaryActor(b Blockchain, localAcc *wallet.Account, committee keys.PublicKeys) (*notary.Actor, error) {
+	committeeMultiSigM := smartcontract.GetMajorityHonestNodeCount(len(committee))
+	committeeMultiSigAcc := wallet.NewAccountFromPrivateKey(localAcc.PrivateKey())
+
+	err := committeeMultiSigAcc.ConvertMultisig(committeeMultiSigM, committee)
+	if err != nil {
+		return nil, fmt.Errorf("compose committee multi-signature account: %w", err)
+	}
+
+	return notary.NewActor(b, []actor.SignerAccount{
+		{
+			Signer: transaction.Signer{
+				Account: localAcc.ScriptHash(),
+				Scopes:  transaction.CalledByEntry,
+			},
+			Account: localAcc,
+		},
+		{
+			Signer: transaction.Signer{
+				Account: committeeMultiSigAcc.ScriptHash(),
+				Scopes:  transaction.CalledByEntry,
+			},
+			Account: committeeMultiSigAcc,
+		},
+	}, localAcc)
+}
+
+// notaryDepositDeficiencyHandler is a function returned by initNotaryDepositDeficiencyHandler.
+// True argument is passed when there is not enough GAS on local account's
+// balance in the Notary contract, false - when local account's Notary deposit
+// expires before particular fallback transaction.
+//
+// The function is intended to be called multiple times on each deposit problem
+// encounter. On each call, It attempts to fix Notary deposit problem without
+// waiting for success. Caller should by default wait for the problem to be
+// fixed, and if not, retry.
+//
+// notaryDepositDeficiencyHandler must not be called from multiple routines in
+// parallel.
+type notaryDepositDeficiencyHandler = func(lackOfGAS bool)
+
+// Amount of GAS for the single local account's GAS->Notary transfer. Relatively
+// small value for fallback transactions' fees.
+var singleNotaryDepositAmount = big.NewInt(int64(fixedn.Fixed8FromInt64(1))) // 1 GAS
+
+// constructs notaryDepositDeficiencyHandler working with the specified
+// Blockchain and GAS/Notary balance of the given account.
+func initNotaryDepositDeficiencyHandler(l *zap.Logger, b Blockchain, monitor *blockchainMonitor, localAcc *wallet.Account) (notaryDepositDeficiencyHandler, error) {
+	_actor, err := actor.NewSimple(b, localAcc)
```

Comment:
s/`_actor`/`act`

Or choose any other nice name, it's not the C#.

Reply:
to me `_actor` is a nice name, and it's a valid Go identifier. Are underscored ids not recommended in Go? ref me to spec or effective go if so

Reply:
ok, lets replace it. But i don't think `act` is good since it is a verb and more appropriate for function. How about `actr`/`simpleActor`/`localActor`?

Reply:
fixed